File: 0605 can-place-flowers/solution.cpp
class Solution {
public:
    bool canPlaceFlowers(vector<int>& f, int n) {
        for(int i=0;i<f.size();i++) {
            if(f[i]==0) {
                bool left = (i==0) || (f[i-1] == 0);
                bool right = (i==f.size()-1) || (f[i+1] == 0);
                if(left && right) {
                    f[i]=1;
                    n--;
                    if(n==0) return true;
                }
            }
        }
        return n<=0;
    }
};

File: 0605 can-place-flowers/solution.rs
impl Solution {
    pub fn can_place_flowers(mut f: Vec<i32>, mut n: i32) -> bool {
        for i in 0..f.len() {
            if f[i] == 0 {
                let left = (i==0) || (f[i-1]==0);
                let right = (i==f.len()-1) || (f[i+1]==0); 
                if left && right {
                    f[i] = 1;
                    n-=1;
                    if n==0 {
                        return true
                    }
                }
            }
        }
        n<=0
    }
}
